### ISSUE TITLE
Ordered comparison between pointer and zero ('char *' and 'int') error

### ISSUE THREAD

**Issue by iThinker** [tat/mimetic#5]
Clang 4 produces this error in mmfile.cxx line 60.

**Comment by teusbenschop** [comment]
The same error occurs here on macOS High Sierra.

```
mmfile.cxx:60:14: error: ordered comparison between pointer and zero ('char *' and 'int')
```

The line 
```if(m_beg > 0)```
was changed to 
```if(m_beg > (char *)0)```
and now it compiles.

**Comment by fake666** [comment]
i think checking for nullptr is nicer. i didn't bother creating a pull request, fix:

https://github.com/fake666/mimetic/commit/34c7e9563cbcab004d43d9a150361ca10715cc5a

merged in my fork

**Comment by teusbenschop** [comment]
Thanks man!

**Comment by radoslawkojdecki** [comment]
man mmap:

"Upon successful completion, mmap() returns a pointer to the mapped region.  Otherwise, a value of MAP_FAILED is returned and errno is set to indicate the error. "

MAP_FAILED is mapped to -1, not nullptr !!!

**Comment by heirecka** [comment]
gcc11 started to produce an error about that, too:
```
mmfile.cxx: In member function 'bool mimetic::MMFile::map()':
mmfile.cxx:60:14: error: ordered comparison of pointer with integer zero ('char*' and 'int')
   60 |     if(m_beg > 0)
      |        ~~~~~~^~~
```

**Comment by heirecka** [comment]
After writing the comment above I realized that the problem is already fixed by bf84940f. So, sorry for the noise and I guess this issue can be closed.

**Comment by tat** [comment]
yeah, thanks for looking into it anyway!
